packages/ui/src/common/components/ModalLink.tsx

```diff
@@ -0,0 +1,30 @@
+import React from 'react'
+import { ReactNode } from 'react-markdown'
+import { Link } from 'react-router-dom'
+
+import { MemberModalCall } from '@/memberships/components/MemberProfile'
+
+import { useModal } from '../hooks/useModal'
+import { AnyModalCall } from '../providers/modal/types'
+
+interface ModalLinkProps<Call> {
+  call: Call extends AnyModalCall ? Call : never
+  children: ReactNode
+}
+
+const ModalLink = ({ call, children }: ModalLinkProps<AnyModalCall>) => {
+  const { showModal } = useModal()
+  return (
+    <Link
+      to="#"
+      onClick={(evt) => {
+        evt.preventDefault()
+        showModal(call)
+      }}
+    >
+      {children}
+    </Link>
+  )
+}
+
+export const MemberModalLink = (props: ModalLinkProps<MemberModalCall>) => ModalLink(props)
```

Comment:
We shouldn't introduce domain-specific components inside general ones. This should be removed from this file.

Reply:
You can also have something like this:

```jsx
const MemberModalLink: IModalLink<MemberModalCall> = ModalLink
```

Reply:
The reason to have an "alias" as above is that it does not create another function ([TS playground](https://www.typescriptlang.org/play?#code/AQSwdgLgpgTgZgQwMZWAWQPYBMEBsDCeuAPGgHzADeAsAFDDAC22eAXOnQL512iSyIU6FrgDqICAAsAIgggJCuEmgA0waRSgAPaGCwBnYTgJFSFGvWA557aVx70IATwAOqAIJgnmY4tzAAXiM8P2J9CBhwAHMKAB9gsQkZOQVTcMiwKLUwKAA3WDI6cGh4ZFQfPAAZcABrAAUYDBd9Yj9zXmAkInY-YG1dA2BPbxFegH5gXvYc-JgOpEkQXCwYKDB2BC97WjpnN2AAMXxiABUKIIAKF2BWE4BKQIpcjBAsB2A91ABJCtxqsBqrSI50Ox1+-3qjWaQKUZEKO1oSAwYHCCQhgWAMP8-TWg2GvzaV3Y4NqDSaLTaDwCFAArnooHBwFA3gjPugoIwAEawAlEDG-cRSWTyUIAcjQHO5MFFakooCw7HS0WAnHhdAA9OqhoY8CAEPo6EiURB2VyeSIIewfhbaqRJebfMD+TaAQ5NdrgAgYI0AO7AOB0pASZGG5GoiVmmAkgHufS9S4uKH6YkuyHku2R3mwqkUaM1K5Jh5AA)).